static/app/utils/profiling/profile/continuousProfile.tsx

```diff
@@ -0,0 +1,187 @@
+      while (start >= 0) {
+        if (framesInStack[start].frame === node.frame) {
+          // The recursion edge is bidirectional
+          framesInStack[start].recursive = node;
+          node.recursive = framesInStack[start];
+          break;
+        }
+        start--;
+      }
+
+      framesInStack[i] = node;
+    }
+
+    node.selfWeight += duration;
+    this.minFrameDuration = Math.min(duration, this.minFrameDuration);
+
+    // Lock the stack node, so we make sure we dont mutate it in the future.
+    // The samples should be ordered by timestamp when processed so we should never
+    // iterate over them again in the future.
+    for (const child of node.children) {
+      child.lock();
+    }
+
+    node.frame.selfWeight += duration;
+
+    for (const stackNode of framesInStack) {
+      stackNode.frame.totalWeight += duration;
+      stackNode.count++;
+    }
+
+    // If node is the same as the previous sample, add the weight to the previous sample
+    if (node === this.samples[this.samples.length - 1]) {
+      this.weights[this.weights.length - 1] += duration;
+    } else {
+      this.samples.push(node);
+      this.weights.push(duration);
+    }
+  }
+
+  // @TODO implement this when we need to extend time ranges and append new profiles
+  appendProfileStart(): ContinuousProfile {
+    throw new Error('Not implemented');
+  }
+  // @TODO implement this when we need to extend time ranges and append new profiles
+  appendToProfileEnd(): ContinuousProfile {
+    throw new Error('Not implemented');
+  }
+
+  build(): ContinuousProfile {
+    this.duration = Math.max(
+      this.duration,
+      this.weights.reduce((a, b) => a + b, 0)
+    );
+
+    // We had no frames with duration > 0, so set min duration to timeline duration
+    // which effectively disables any zooming on the flamegraphs
+    if (
+      this.minFrameDuration === Number.POSITIVE_INFINITY ||
+      this.minFrameDuration === 0
+    ) {
+      this.minFrameDuration = this.duration;
+    }
+
+    return this;
+  }
+}
+
+const COCOA_MAIN_THREAD = 'com.apple.main-thread';
+function getThreadData(profile: Profiling.SentryContinousProfileChunk): {
+  threadId: number;
+  threadName: string;
+} {
+  const {samples, queue_metadata = {}, thread_metadata = {}} = profile.profile;
+  const sample = samples[0];
+  const threadId = parseInt(sample.thread_id, 10);
+  const threadName = thread_metadata?.[threadId]?.name;
+
+  if (threadName) {
+    return {threadId, threadName};
+  }
+
+  // cocoa has a queue address that we fall back to to try to get a thread name
+  // is this the only platform string to check for?
+  if (profile.platform === 'cocoa') {
+    const queueName =
+      sample.queue_address && queue_metadata?.[sample.queue_address]?.label;
+
+    // if a queue has the main thread name, we discard it
+    if (queueName && queueName !== COCOA_MAIN_THREAD) {
+      return {threadId, threadName: queueName};
+    }
+  }
```

Comment:
Mentioned this earlier but I don't think we kept the queue names for continuous profiling. It might be mixed together with the thread data so maybe let's double-check with @armcknight